cmd/k8s-operator/operator.go

```diff
@@ -133,6 +134,14 @@ func main() {
 			}
 		}()
 	}
+
+	// Operator log uploads can be opted-out using the "TS_NO_LOGS_NO_SUPPORT" environment variable.
+	if !envknob.NoLogsNoSupport() {
+		zlog = zlog.WithOptions(zap.WrapCore(func(core zapcore.Core) zapcore.Core {
```

Comment:
Instead of creating a whole new core, could we just swap out the `io.Writer` that the original core writes to? That would make things much simpler and avoid the need to serialise each log twice. I had a little go locally and it needs a bit of smithing/reordering but in broad terms this approach seemed to work afaict:

```go
type teeWriter struct {
	logf logger.Logf
}

func (t *teeWriter) Write(p []byte) (n int, err error) {
	t.logf("%s", p)
	return os.Stderr.Write(p)
}

tee := &teeWriter{logf: s.Logtailf()}
opts := []kzap.Opts{
	kzap.WriteTo(tee),
}
```

Reply:
I'd be happy with either:

1. Accept that limitation - it's scope creep to always include debug logs compared to what we set out to do, and most of the value is having the reconciler logs at all
2. Or if debug logging isn't enabled for stderr, create a second core that has debug enabled but only writes to logtailf, and zap.Tee to that. It's a bit more complex and we'll serialise logs twice, but at least we don't need the code for our own zap core implementation.

WDYT? I kind of prefer option 1 tbh

Reply:
There's probably also a third option where we create a debug zap core as the root core which has its `io.Writer` set to logtailf, and zap.Tee that to a separate stderr core that respects the configured level setting? Dunno though, I'd need to have a play with it in a code editor to work it out.

Reply:
With a quick search we have ~150 calls to .Info and ~190 calls to .Debug. I'd be worried that we'll lose a lot of context by tying the log level to what ends up in logz.

Reply:
Looks great! That's a really nice improvement 👍